Bmx.Core.Test/EventTests.cs

```diff
@@ -0,0 +1,218 @@
+using System;
```

Comment:
Double-check: are more tests inbound? I noticed we don't verify calls to `_identityProvider`, and it doesn't seem like the conditionals in https://github.com/Brightspace/bmx/blob/46f5346c6b1be35e158d8366bedaf96b43980c07/Bmx.Core/BmxCore.cs#L71-L118 are being covered.

Reply:
>  I noticed we don't verify calls to _identityProvider
I have tests written for testing the methods inside Bmx.Idp.Okta (separate test project). https://github.com/Brightspace/bmx/pull/208 (its WIP). Is 

that what you meant? Here Bmx.Core is tested in isolation by using a mock Identity Provider.

> it doesn't seem like the conditionals in [Line 71 - 118] are being covered.
Yeah :( tests for these are bit more complicated so I was planning to leave them out till later.

As an aside, it seems like a bit of a warning the code is doing a lot more than it should in one go when testing it is hard.

Reply:
> > I noticed we don't verify calls to `_identityProvider`
> 
> I have tests written for testing the methods inside Bmx.Idp.Okta (separate test project). #208 (its WIP).
> 
> Is that what you meant? Here Bmx.Core is tested in isolation by using a mock Identity Provider.

Sort of. `_identityProvider` is an implementation of `IIdentityProvider<TAuthenticateState, TAuthenticatedState, TAccountState>`. We can't test the implementation here, but we can test that we are calling the interface methods properly.

For example, we can confirm that `_identityProvider.Authenticate()` gets called with the proper arguments (from https://github.com/Brightspace/bmx/blob/46f5346c6b1be35e158d8366bedaf96b43980c07/Bmx.Core/BmxCore.cs#L54-L55).

If you're convinced to switch to strict mocks and call `_mockIdentityProvider.VerifyAll()` at the end of every test (maybe on `[Teardown]`), that should help verify most of the calls to `_identityProvider` if you only set up for arguments you expect.